ui/lib/apps/InstanceProfiling/pages/Detail.tsx

```diff
@@ -37,6 +69,100 @@ function isFinished(data) {
   return groupState === 2 || groupState === 3
 }
 
+async function getActionToken(
+  id: string,
+  apiType: string
+): Promise<string | undefined> {
+  const res = await client.getInstance().getActionToken(id, apiType)
+  const token = res.data
+  if (!token) {
+    return
+  }
+  return token
+}
+
+function ViewResultButton({ rec, t }) {
+  const isProtobuf: boolean =
+    rec.profile_raw_data_type === ProfilingRawDataTypeProtobuf
+  let token: string | undefined
+
+  const handleViewResultMenuClick = usePersistFn(async (e: MenuInfo) => {
+    switch (e.key) {
+      case 'download':
+        token = await getActionToken(rec.id, 'single_download')
+
+        window.location.href = `${client.getBasePath()}/profiling/single/download?token=${token}`
```

Comment:
What scenario will token be empty? Maybe we shouldn't download or open a new window when token is empty?